app/logic/getUpcomingEvents.py

```diff
@@ -0,0 +1,27 @@
+from app.models.program import Program
+from app.models.interest import Interest
+from app.models.event import Event
+from app.models.user import User
+from datetime import date, datetime, time
+
+def getUpcomingEventsForUser(user):
+
+    interestedEvent = (Event.select(Event, Program, Interest)

```

Comment:
This name should be plural, since it's a list (sort of)

Reply:
You also can simplify the query a little. You don't need to specify anything in the select (I'm pretty sure), which means you don't need `.objects()` down below. And since you have a program object you are passing through, your jinja can use that for the name, rather than needing to select programName from the query explicitly.